tests/integration/pilot/common/routing.go

```diff
@@ -321,6 +320,10 @@ func protocolSniffingCases(apps *EchoDeployments) []TrafficTestCase {
 				destinations := destinations
 				// grabbing the 0th assumes all echos in destinations have the same service name
 				destination := destinations[0]
+				if apps.Headless.Contains(client) || apps.Headless.Contains(destinations) && len(apps.Headless) > 1 {
```

Comment:
```suggestion
				if (apps.Headless.Contains(client) || apps.Headless.Contains(destinations)) && len(apps.Headless) > 1 {
```

If I understand the goal right.